### PR TITLE
Initialized project

### PR DESCRIPTION
**Pull request by DarkLordImperatus**
Added Login controller and model
Added Pseudocode for SQL logic
Added DBconnection handler for future MySQL db
Added Existing User Model and interface
Added ServerResponse model for future implementation of all responses/error handling
Updated version attributes to properly compile
Added MySQL reference

### CODE REVIEW

**Comment by DarkLordImperatus** [comment]
.

File: CGen.Core.Api/Controllers/HubSpotController.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace CGen.Core.Api.Controllers
+{
+    [ApiVersion("1.0")]
+    [Produces("application/json")]
+    [Authorize(Roles = "web-unauth")]
+    [Route("api/v{ver:apiVersion}/[controller]")]
+    public class HubSpotController : Controller
```
Comment (cgen01) [inline CGen.Core.Api/Controllers/HubSpotController.cs]: This controller is not needed in this project.

File: CGen.Core.Api/Controllers/PaymentsController.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CGen.Core.Api.Controllers
+{
+    [ApiVersion("1.0")]
+    [Authorize(Roles = "payments")]
+    [Route("api/v{ver:apiVersion}/[controller]")]
+    public class PaymentsController : Controller
```
Comment (cgen01) [inline CGen.Core.Api/Controllers/PaymentsController.cs]: This controller is not needed in this project.

File: CGen.Core.Api/Properties/PublishProfiles/WebDeployQa.pubxml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (cgen01) [inline CGen.Core.Api/Properties/PublishProfiles/WebDeployQa.pubxml]: This file should be removed.

File: CGen.Core.Api/appsettings.Development.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿{
+  "Logging": {
+    "IncludeScopes": false,
+    "LogLevel": {
+      "Default": "Debug",
+      "System": "Information",
+      "Microsoft": "Information"
+    }
+  },
+  "HubSpot": {
```
Comment (cgen01) [inline CGen.Core.Api/appsettings.Development.json]: The HubSpot configuration can be removed.

File: CGen.Core.Api/appsettings.Development.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "Url": "https://api.hubapi.com",
+    "Key": "74f82c7a-87d1-4b30-99bd-1ee2f67c9b82"
+  },
+  "Jwt": {
+    "Key": "HCXdiNCBRpVvaUsWLoibHYNeGvs39C4bVhMpHatawpyjRc7Br1fSh9H1aC9m6KtBnBr6muzLxG4XtBveP05TcAXIKGYErm56",
+    "Issuer": "http://localhost:5000/",
+    "Audience": "candee-generations"
+  },
+  "ConnectionStrings": {
+    "CGenConnection": "Data Source=tcp:s18.winhost.com;Initial Catalog=DB_114230_cgentest;User ID=DB_114230_cgentest_user;Password=KZAazYLtinp1;Integrated Security=False;"
```
Comment (cgen01) [inline CGen.Core.Api/appsettings.Development.json]: This connection string should be removed/replaced with your connection string.

File: CGen.Core.Api/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        "Default": "Warning"
+      }
+    },
+    "Console": {
+      "LogLevel": {
+        "Default": "Warning"
+      }
+    }
+  },
+  "HubSpot": {
```
Comment (cgen01) [inline CGen.Core.Api/appsettings.json]: The HubSpot configuration should be removed.

File: CGen.Core.Api/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    "Url": "https://api.hubapi.com",
+    "Key": "74f82c7a-87d1-4b30-99bd-1ee2f67c9b82"
+  },
+  "Jwt": {
+    "Key": "HCXdiNCBRpVvaUsWLoibHYNeGvs39C4bVhMpHatawpyjRc7Br1fSh9H1aC9m6KtBnBr6muzLxG4XtBveP05TcAXIKGYErm56",
+    "Issuer": "https://candeegenerations.com/api/",
+    "Audience": "candee-generations"
+  },
+  "ConnectionStrings": {
+    "CGenConnection": "Data Source=tcp:s18.winhost.com;Initial Catalog=DB_114230_cgentest;User ID=DB_114230_cgentest_user;Password=KZAazYLtinp1;Integrated Security=False;"
```
Comment (cgen01) [inline CGen.Core.Api/appsettings.json]: This connection string should be removed/replaced with your connection string.

File: CGen.Core.Domain/DomainObjects/Client.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using CGen.Core.Domain.Common;
+
+namespace CGen.Core.Domain.DomainObjects
+{
+    public class Client : GuidId
+    {
```
Comment (cgen01) [inline CGen.Core.Domain/DomainObjects/Client.cs]: I don't think this is needed for this project.

File: CGen.Core.Domain/DomainObjects/Invoice.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using CGen.Core.Domain.Common;
+
+namespace CGen.Core.Domain.DomainObjects
+{
+    public class Invoice : GuidId
+    {
```
Comment (cgen01) [inline CGen.Core.Domain/DomainObjects/Invoice.cs]: This is not needed for this project.

File: CGen.Core.Domain/DomainObjects/Payment.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using CGen.Core.Domain.Common;
+
+namespace CGen.Core.Domain.DomainObjects
+{
+    public class Payment : GuidId
+    {
```
Comment (cgen01) [inline CGen.Core.Domain/DomainObjects/Payment.cs]: This is not needed for this project.

File: CGen.Core.Domain/Models/HubSpotContactModel.cs [hunk 11]
```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+
+namespace CGen.Core.Domain.Models
+{
+    public class HubSpotContactModel
+    {
```
Comment (cgen01) [inline CGen.Core.Domain/Models/HubSpotContactModel.cs]: This is not needed for this project.

File: CGen.Core.Repository/Context/CGenContext.cs [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.EntityFrameworkCore;
+
+namespace CGen.Core.Repository.Context
+{
+    public class CGenContext : DbContext
+    {
+        public CGenContext(DbContextOptions options) : base(options)
+        { }
+
+        public DbSet<Payment> Payments { get; set; }
```
Comment (cgen01) [inline CGen.Core.Repository/Context/CGenContext.cs]: This can be removed.

File: CGen.Core.Repository/Migrations/20180410234353_InitialCreate.cs [hunk 13]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
Comment (cgen01) [inline CGen.Core.Repository/Migrations/20180410234353_InitialCreate.cs]: All of these migration files should be removed.

**Pull request merged**